[FIX] web_editor: correctly check the toggle of new window link

rdeodoo opened the pull request:
Before this commit and most likely since commit [1], the "Open in new window" option of link would not persist through editions.

Steps to reproduce:
- Create a link in the website builder
- Set its "Open in new window" property to true
- Save, and see that it works as expected
- Now enter edit mode again, the toggle will be off and not on as it should be since the `target="_blank"` was correctly added.

For easy of debugging and understanding, you can just grep this line
```js
this.initialNewWindow = this.initialNewWindow || this.linkEl.target === '_blank';
```

[1]: https://github.com/odoo/odoo/commit/d7245d2abf528d093226c80e40975e63d61e8997#

opw-3781477

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/156372).

rdeodoo commented:
@odoo/rd-runbot I changed the target of the PR and rebased the branch but it seems like it's not working this time?
Runbot build still think this targets 17

rdeodoo commented:
@odoo/rd-framework-editor Small one, could you have a look?

dmo-odoo commented:
@robodoo r+
